hbase-client/src/main/java/org/apache/hadoop/hbase/client/MetricsConnection.java

```diff
@@ -295,8 +337,13 @@ public Counter newMetric(Class<?> clazz, String name, String scope) {
     }
   };
 
+  // List of thread pool per connection of the metrics.
+  private List batchPools = new ArrayList<Supplier>();
+  private List metaPools = new ArrayList<Supplier>();
```

Comment:
Is a typed list possible here?
```suggestion
  private List<Supplier<ThreadPoolExecutor>> batchPools = new ArrayList<>();
  private List<Supplier<ThreadPoolExecutor>> metaPools = new ArrayList<>();
```

Reply:
Good suggestion, updated, thanks!